docs/VnVPlan/VnVPlan.tex

```diff
@@ -209,55 +209,488 @@ \section{System Test Description}
+Input: API calls from client (front-end) to back-end containing receipt data.
+          
+Output: Database contains correctly organized items.
+
+Test Case Derivation: System back-end must accurately map and sort receipt entries provided, keeping a record of each item all users purchase.
+          
+How test will be performed: Begin with an empty testing database. Test cases comprised of items and prices are sent through an API call to the system back-end by testing client. Then an additional API call will be called from the testing client against the system back-end to validate that the items and prices are correctly stored in the database.
+
+\item{\textbf{FRT-FR7-2}}
+
+Control: Automatic, Dynamic
+          
+Initial State: Test database is populated.
+
+Input: API calls from client (front-end) to back-end containing receipt data.
+          
+Output: Database contains correctly organized items.
+
+Test Case Derivation: System back-end must accurately map and sort receipt entries provided, keeping a record of each item all users purchase.
+          
+How test will be performed: Begin with a populated testing database. Test cases comprised of items and prices are sent through an API call to the system back-end by testing client. Then an additional API call will be called from the testing client against the system back-end to validate that the items and prices are correctly stored in the database.
+
+\item{\textbf{FRT-FR7-3}}
+
+Control: Automatic, Dynamic
+          
+Initial State: Test database is populated.
+
+Input: API calls from client (front-end) to back-end containing receipt data. Data containing items that are the same as existing items in the database but with slightly different names.
+          
+Output: Database contains correctly organized items and understanding the relationship between like items.
+
+Test Case Derivation: System back-end must accurately map and sort receipt entries provided, keeping a record of each item all users purchase.
+          
+How test will be performed: Begin with a populated testing database. Test cases comprised of items and prices are sent through an API call to the system back-end by testing client. Then an additional API call will be called from the testing client against the system back-end to validate that the items and prices are correctly stored in the database.
+
+\end{enumerate}
+
+\paragraph{Grocery Recommendations}
+
+These tests pertain to all the functional requirements related to the systems grocery recommendations based on grocery data received from user and their preferences.
+
+\begin{enumerate}
+
+\item{\textbf{FRT-FR8-1}}
+
+Control: Manual, Dynamic
+          
+Initial State: User has set budget information.
+
+Input: User submits receipt containing luxury items that are out of budget.
+          
+Output: System provides recommendation to stop purchasing some items that are too expensive and (if available) recommend alternatives that are more in-line with the users budget.
```

Comment:
I think we need to be more specific:

The system should identify the newly purchased item as the cause for the user going over budget and will recommend that the user stop purchasing it. If available, the system will also recommend alternatives that are in-line with the user's budget.